lib/grandpa/grandpa.go

```diff
@@ -321,25 +323,15 @@ func (s *Service) initiateRound() error {
 
 	// make sure no votes can be validated while we are incrementing rounds
 	s.roundLock.Lock()
+	defer s.roundLock.Unlock()
+
 	s.state.round++
 	logger.Debugf("incrementing grandpa round, next round will be %d", s.state.round)
 	s.prevotes = new(sync.Map)
 	s.precommits = new(sync.Map)
 	s.pvEquivocations = make(map[ed25519.PublicKeyBytes][]*SignedVote)
 	s.pcEquivocations = make(map[ed25519.PublicKeyBytes][]*SignedVote)
-	s.roundLock.Unlock()
-
-	best, err := s.blockState.BestBlockHeader()
-	if err != nil {
-		return err
-	}
 
-	if best.Number > 0 {
-		return nil
-	}
-
-	// don't begin grandpa until we are at block 1
-	s.waitForFirstBlock()
```

Comment:
it is not necessary to wait until the first block since the genesis block can be chosen in the next rounds if there are no new blocks, so hanging here until we receive/produce a new block is not needed